torch/csrc/distributed/rpc/rpc_agent.cpp

```diff
@@ -6,16 +6,134 @@ namespace rpc {
 
 constexpr size_t WorkerInfo::MAX_NAME_LEN;
 
+// TODO(osalpekar): Tune RetryDuration to be something in seconds per
+// team recommendations. Kept at 100ms for the sake of Milestone 1.
+const auto kRpcRetryDuration = std::chrono::milliseconds(100);
```

Comment:
It seems like if we have don't have an end-to-end response in 100ms, we'll immediately retry, right?

If we ever have 100ms of queuing on a server side, or any hiccup, or too much GIL waiting, or  TCP hiccup, we'll start a retry storm, right?

(in addition 1.2 is fairly aggressive as a backoff, maybe 2.0 [or even 1.5] is more appropriate)

I'm wondering if, policy-wise, we might consider:
  - if we get an immediate connection-broken message, we retry right away (which doesn't seem to be the case below)... e.g. if we see TCP FIN on the socket, there's no sense in waiting.
  - having a somewhat longer timeout, if we hear nothing